tools/rosconsole/src/rosconsole/logger_level_service_caller.py

```diff
@@ -0,0 +1,141 @@
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+import rosnode
+import rospy
+import rosservice
+# We'll resolve names via rosgraph.names
+import rosgraph
+
+
+class ROSConsoleException(Exception):
+    """
+    Base exception class of rosconsole-related errors
+    """
+    pass
+
+
+class LoggerLevelServiceCaller(object):
+    """
+    Handles service calls for getting lists of nodes and loggers
+    Also handles sending requests to change logger levels
+    """
+    def __init__(self):
+        pass
+
+    def get_levels(self):
+        # Declare level names lower-case, because that's how they are returned
+        # from the service call.
+        return ['debug', 'info', 'warn', 'error', 'fatal']
+
+    def get_loggers(self, node):
+        self._refresh_loggers(node)
+        return self._current_loggers
+
+    def get_node_names(self):
+        """
+        Gets a list of available services via a ros service call.
+        :returns: a list of all nodes that provide the set_logger_level service, ''list(str)''
+        """
+        set_logger_level_nodes = []
+        nodes = rosnode.get_node_names()
+        for name in sorted(nodes):
+            for service in rosservice.get_service_list(name):
+                if service == name + '/set_logger_level':
+                    set_logger_level_nodes.append(name)
+        return set_logger_level_nodes
+
+    def _refresh_loggers(self, node):
+        """
+        Stores a list of loggers available for passed in node
+        :param node: name of the node to query, ''str''
+        :raises: :exc:`ROSTopicException` If topic type cannot be determined or loaded
+        """
+        self._current_loggers = []
+        self._current_levels = {}
+        # Construct the service name, taking into account our namespace
+        servicename = rosgraph.names.ns_join(
+            rosgraph.names.ns_join(rosgraph.names.get_ros_namespace(), node),
+            'get_loggers')
+        # Construct the service name, taking into account our namespace
+        servicename = rosgraph.names.resolve_name(
+            servicename, rosgraph.names.get_ros_namespace())
+        try:
+            service = rosservice.get_service_class_by_name(servicename)
+        except rosservice.ROSServiceException as e:
+            raise ROSConsoleException("node %s doesn't exist or doesn't support query" % node)
```

Comment:
Please quote the variable output with single quote for better readability: 

    "node '%s' doesn't exist or doesn't support query" % node